### PR TITLE
Added read_user_input_validate and input_static_dynamic 

### PR DESCRIPTION
**Pull request by ritika2**
Added 2 function for release 2
input_static_dynamic
read_user_input_validate

### CODE REVIEW

File: src/read_user_input_validate.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    while((scanf("%d",&passenger_id))==0){
+    	//To check that user enters an integer value only
+    		scanf("%*[^\n]");
+    		printf("Invalid entry, please enter a valid number: ");
+    }
+    if(passenger_id <=0){
+        	printf("Invalid passenger id entered");
+        	exit(1);
+    }
+    printf("Enter the arrival floor between 1 and %d:",elevator_count);
```
Comment (MaazJamal) [inline src/read_user_input_validate.c]: This should be `%d:",floor_count);`

File: src/read_user_input_validate.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    while((scanf("%d",&arrival_floor))==0){
+    	//To check that user enters an integer value only
+        	scanf("%*[^\n]");
+        	printf("Invalid entry, please enter a valid number: ");
+    }
+    if(arrival_floor<=0 || arrival_floor > elevator_count){
```
Comment (MaazJamal) [inline src/read_user_input_validate.c]: `arrival_floor > floor_count)`

File: src/read_user_input_validate.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    		printf("Invalid arrival floor entered!!");
+    		exit(1);
+    }
+    printf("Enter the destination floor between 1 and %d:",elevator_count);
```
Comment (MaazJamal) [inline src/read_user_input_validate.c]: `and %d:",floor_count);`

File: src/read_user_input_validate.c [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    while((scanf("%d",&dest_floor))==0){
+    	//To check that user enters an integer value only
+        	scanf("%*[^\n]");
+        	printf("Invalid entry, please enter a valid number: ");
+    }
+    if(dest_floor <=0 || dest_floor > elevator_count){
```
Comment (MaazJamal) [inline src/read_user_input_validate.c]: if(dest_floor <=0 || dest_floor > floor_count)

**Review by MaazJamal — CHANGES_REQUESTED** [review]
global variable elevator_count needs to be replaced with global variable floor_count

File: src/read_user_input_validate.c (hunk 4 above)
Comment (ritika2) [reply src/read_user_input_validate.c]: > global variable elevator_count needs to be replaced with global variable floor_count

Modified the file

File: src/read_user_input_validate.c [hunk 5]
```diff
@@ -38,25 +38,25 @@ void read_user_input_validate(struct passenger *passenger_info, unsigned int tim
         	printf("Invalid passenger id entered");
         	exit(1);
```
Comment (MaazJamal) [inline src/read_user_input_validate.c]: it shouldnt exit the program when the user enters invalid input. It should ask again. Something like this should work. The rest is fine. 
```
    int correct = 0;
    while( correct == 0){
          while((scanf("%d",&arrival_floor))==0){
    	        //To check that user enters an integer value only
        	scanf("%*[^\n]");
        	printf("Invalid entry, please enter a valid number: ");
        }
        if(arrival_floor<=0 || arrival_floor > elevator_count){
        if(arrival_floor<=0 || arrival_floor > floor_count){
    		printf("Invalid arrival floor entered!!");
    		correct == 0;
                continue;
         }
          correct = 1;
    }
```

**Review by MaazJamal — CHANGES_REQUESTED** [review]

**Review by Hariprasadm1408 — APPROVED** [review]

File: src/read_user_input_validate.c (hunk 5 above)
Comment (ritika2) [reply src/read_user_input_validate.c]: Fixed

**Review by MaazJamal — APPROVED** [review]
Looks good. Thanks.

**Review by divyalakshmi-padmanaban — APPROVED** [review]

**Pull request merged**